subsys/net/lib/wifi_mgmt_ext/wifi_mgmt_ext.c

```diff
@@ -24,6 +24,60 @@ LOG_MODULE_REGISTER(wifi_mgmt_ext, CONFIG_WIFI_MGMT_EXT_LOG_LEVEL);
 		     "CONFIG_WIFI_CREDENTIALS_STATIC_SSID required");
 #endif /* defined(CONFIG_WIFI_CREDENTIALS_STATIC) */
 
+static int __stored_creds_to_params(struct wifi_credentials_personal *creds,
+				struct wifi_connect_req_params *params)
+{
+	char *ssid = NULL;
+	char *psk = NULL;
+
+	/* SSID */
+	ssid = (char *)k_malloc(creds->header.ssid_len + 1);
+	if (!ssid) {
+		LOG_ERR("Failed to allocate memory for SSID\n");
+		return -ENOMEM;
+	}
+	memset(ssid, 0, creds->header.ssid_len + 1);
+	strncpy(ssid, creds->header.ssid, creds->header.ssid_len);
+	params->ssid = ssid;
+	params->ssid_length = creds->header.ssid_len;
+
+	/* PSK (optional) */
+	if (creds->password_len > 0) {
+		psk = (char *)k_malloc(creds->password_len + 1);
```

Comment:
use `k_zalloc` to avoid extra memset.

Reply:
Looks like k_zalloc is not available in zephyr. Keeping this as is.